### DISCUSSION TITLE
Speed and or search problems

### DISCUSSION THREAD

**Discussion by esgheorghe** [Ylianst/MeshCentral#6579] [Issues]
Hello,

We have encountered the following problem with MC, maybe someone has an idea?

The performance when searching or clicking in the different areas of MeshCentral is very sluggish or even no longer manageable 
as soon as a few thousand users (MeshCentral Windows Agents) are visible in MC.

The maxDeviceView setting has brought an improvement, but if you search for a user (u:xxxx) only the first page is searched and the other 300 pages are ignored.

Our MeshCentral setup looks like this:

MeshCentral - Version: 1.1.35 
OS: RedHat Enterprise Linux Server 9.5
Database: MongoDB.
Node - Version: v22.11.0

The server is virtualized in ESX-VM and has 32GB RAM, 8 cores and 10GB network connection. 
User authentication and creation is controlled via ActiveDirectory and works perfectly.

The MC is operated behind an Ngnix SSL proxy.

Config:
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "settings": {
    "MongoDb": "mongodb://xxxUser:xxxPWDj@127.0.0.1:27017/db_meshcentral",
    "MongoDbcol": "meshcentral",
    "cert": "myserver.mydomain.com",
    "syslog": "meshcentral",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 4430,
    "aliasPort": 443,
    "redirPort": 800,
    "_redirAliasPort": 80,
    "agentLogDump": true,
    "agentCoreDump": true,
    "trustedProxy": "10.xx.xx.xx,127.0.0.1,srv_xxxxx,srv_xxxxx.internal.domain",
    "TlsOffload": "10.xx.xx.xx,127.0.0.1,srv_xxxxx,srv_xxxxx.internal.domain,CloudFlare",
    "WebRTC": true,
    "AllowHighQualityDesktop": true,
    "agentTimeStampServer": false,
    "MaxInvalidLogins": {
                "time": 10,
                "count": 3,
                "coolofftime": 60
          },
    "newAccounts": false
  },
  "domains": {
    "": {
      "title": "xxxx IT",
      "_title2": "https://myserver.mydomain.com",
      "_minify": true,
      "mstsc": true,
      "_maxDeviceView": 50,
      "_siteStyle": 3,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "certUrl": "https://myserver.mydomain.com",
      "auth": "ldap",
            "ldapoptions": {
                "url": [
                    "ldap://srv_xxxxx.internal.domain:389",
                    "ldap://srv_xxxxx.internal.domain:389"
                ],
                "bindDN": "CN=Internal User",
                "bindCredentials": "i dont know",
                "searchBase": "DC=internal,DC=domain",
                "searchFilter": "(sAMAccountName={{username}})"
            },
            "ldapsiteadmingroups": "CN=SEC_MeshCentral_Admin,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
            "ldapuserrequiredgroupmembership": [
                "CN=SEC_MeshCentral_Admin,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
                "CN=SEC_MeshCentral_Citrix,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
                "CN=SEC_MeshCentral_Clients,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
                "CN=SEC_MeshCentral_GL,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
                "CN=SEC_MeshCentral_POS,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
                "CN=SEC_MeshCentral_Server,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
                "CN=SEC_MeshCentral_Wallboards,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain",
                "CN=SEC_MeshCentral_MacOS,OU=MeshCentral,OU=Groups,OU=OU,DC=internal,DC=domain"
            ],
            "ldapsyncwithusergroups": {
                "filter": [
                    "SEC_MeshCentral_Admin",
                    "SEC_MeshCentral_Citrix",
                    "SEC_MeshCentral_Clients",
                    "SEC_MeshCentral_GL",
                    "SEC_MeshCentral_POS",
                    "SEC_MeshCentral_Server",
                    "SEC_MeshCentral_Wallboards",
                    "SEC_MeshCentral_MacOS"
                ]
            },
            "_authStrategies": {
                "_azure": {
                    "_clientid": "SECRET",
                    "_clientsecret": "SECRET",
                    "_tenantid": "SECRET",
                    "_newAccounts": true
                }
            },
            "consentMessages": {
                "Title": "XXXX - RemoteAccess",
                "consentTimeout": 60,
                "autoAcceptOnTimeout": false,
                "desktopnotify": true,
                "terminalnotify": true,
                "filenotify": true,
                "desktopprompt": true,
                "terminalprompt": true,
                "fileprompt": true,
                "desktopprivacybar": true
            },
            "agentTag": {
                "ServerName": 0,
                "ServerDesc": 1,
                "ServerTags": 3
            },
            "hidePowerTimeline": true,
            "userSessionIdleTimeout": 120,
            "userSessionsSort": "Username",
            "ldapusername": "sAMAccountName",
            "ldapUserKey": "sAMAccountName",
            "ldapuseremail": "mail",
            "ldapuserrealname": "{{{givenName}}} {{{sn}}}",
            "ldapuserphonenumber": "telephoneNumber",
            "ldapUserImage": "thumbnailPhoto"
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```

**Comment by si458** [comment]
The search is done in browser not on the server
So if u have 1000+ devices and a weak laptop/PC accessing the web ui then this is going to be a problem
But without access to a setup with 1000+ devices to look at what's happening it's also hard to improve things

**Comment by esgheorghe** [comment]
Hi Simon,

we have various clients from where we access MC.
My laptop is a Dell Precision 7680 (32GB RAM, Core i9 - 13950HX,...).

All tested browsers (Edge, Chrome and Firefox) are x64 and have the problem from approx. 3500 agents / nodes.
Here is the performance analysis of Chrome.
![grafik](https://github.com/user-attachments/assets/bce3c8a6-1fe7-4609-a6eb-d2ea32cb68b5)

![grafik](https://github.com/user-attachments/assets/eab14eea-f356-4361-9f73-2f56508c824f)


Maybe you can optimize the task processing?
https://web.dev/articles/optimize-long-tasks?utm_source=devtools&hl=en

**Reply by si458** [reply]
without access to an environment with those 3500+ clients, we cannot replicate/fix the issue
i myself only have 100 client i look after and i have no issues at all
so IF you are happy, then please email myself and provide access of some form to your environment,
then i can look into whats its doing

**Reply by esgheorghe** [reply]
You got an PM....

**Reply by si458** [reply]
si458 on anything, X, bluesky, discord, github (email on github page)

**Reply by esgheorghe** [reply]
Hello everyone,

I have to raise this topic again.

We still have the problem that the management interface freezes the more nodes we have.

However, we have observed that the freezing of the interface only happens when a new node / user is added. With our Citrix users, this happens every few seconds when a user logs on or logs off.

Our idea would be to implement a refresh timer or manual refresh, so that the supporter / user can decide when the nodes are updated.
Some kind of online / offline switch?

Is this possible or do you have another idea?

Thank you very much
Stefan

**Comment by esgheorghe** [comment]
Hello everyone,

I would like to bring this topic back into focus. 

Even with the latest version, we still have the issue with the WebUI (OLD & NEW Style).

Setting ‘maxDeviceView’ works, and the WebUI is running smoothly again, but the search function (host name, user, etc.) is not working,
as it only searches the current ‘view/page’ and not all of them.
You have to click through every page, which is unreasonable with more than 200 pages.

How could you make the search work across multiple pages, or is there another solution?

Many thanks & best regards,
Stefan
